### PR TITLE
Added BitBucket "default reviewers" to Pull Requests

### PR DESCRIPTION
**Pull request by Jens-H-Eriksen**
I've added the possibility to add "default reviewers" to Pull Requests. 
Default Reviewers are set for each repo in BitBucket.
We really missed this feature as nobody was noticed when new PR was made.

### CODE REVIEW

File: NuKeeper/Collaboration/CollaborationFactory.cs [hunk 1]
```diff
@@ -74,7 +74,7 @@ public class CollaborationFactory : ICollaborationFactory
 
                 if (reader != null)
                 {
-                    _nuKeeperLogger.Normal($"Collaboration platform specified as '{_platform}'");
+                    _nuKeeperLogger.Normal($"Collaboration platform specified as '{reader.Platform}'");
```
Comment (MarcBruins) [inline NuKeeper/Collaboration/CollaborationFactory.cs]: Does `_platform` output a wrong value? Why change this?

File: Nukeeper.BitBucketLocal/BitBucketLocalPlatform.cs [hunk 2]
```diff
@@ -29,8 +30,9 @@ public void Initialise(AuthSettings settings)
             _settings = settings;
             var httpClient = new HttpClient
             {
-                BaseAddress = settings.ApiBase
-            };
+                BaseAddress = new Uri($"{settings.ApiBase.Scheme}://{settings.ApiBase.Authority}")
```
Comment (MarcBruins) [inline Nukeeper.BitBucketLocal/BitBucketLocalPlatform.cs]: Isn't this exactly the same in the end as the old line?

File: Nukeeper.BitBucketLocal/BitbucketLocalRestClient.cs [hunk 3]
```diff
@@ -83,26 +84,26 @@ private async Task<T> HandleResponse<T>(HttpResponseMessage response, [CallerMem
 
         public async Task<IEnumerable<Repository>> GetProjects([CallerMemberName] string caller = null)
         {
-            var response = await GetResourceOrEmpty<IteratorBasedPage<Repository>>("projects?limit=999", caller);
+            var response = await GetResourceOrEmpty<IteratorBasedPage<Repository>>("rest/api/1.0/projects?limit=999", caller);
```
Comment (MarcBruins) [inline Nukeeper.BitBucketLocal/BitbucketLocalRestClient.cs]: Extract this to a const please. I like that it's moved here, that way in the future we can build a new/updated client which target a different rest api version

File: Nukeeper.BitBucketLocal/BitbucketLocalRestClient.cs [hunk 4]
```diff
@@ -111,9 +112,15 @@ public async Task<PullRequest> CreatePullRequest(PullRequest pullReq, string pro
...
-            var response = await _client.PostAsync($@"projects/{projectName}/repos/{repositoryName}/pull-requests", requestBody);
+            var response = await _client.PostAsync($@"rest/api/1.0/projects/{projectName}/repos/{repositoryName}/pull-requests", requestBody);
 
             return await HandleResponse<PullRequest>(response, caller);
         }
+
+        public async Task<IEnumerable<AddReviewer>> GetBitBucketReviewers(string projectName, string repositoryName, [CallerMemberName] string caller = null)
+        {
+            var response = await GetResourceOrEmpty<List<Conditions>>($"rest/default-reviewers/1.0/projects/{projectName}/repos/{repositoryName}/conditions", caller);
+            return response.SelectMany(c => c.Reviewers).Select(usr => new AddReviewer() {User = usr} );
```
Comment (MarcBruins) [inline Nukeeper.BitBucketLocal/BitbucketLocalRestClient.cs]: So who will be the PR reviewer now? Is it the first in the reviewers list?

File: Nukeeper.BitBucketLocal/Models/AddReviewer.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+using Newtonsoft.Json;
+
+namespace NuKeeper.BitBucketLocal.Models
+{
+    public class AddReviewer
```
Comment (shep1987) [inline Nukeeper.BitBucketLocal/Models/AddReviewer.cs]: This class name seems a little strange would `PullRequestReviewer` be better?

The use of `List<AddReviewer>`  in the pull request model is why a change could be good

File: NuKeeper/Collaboration/CollaborationFactory.cs (hunk 1 above)
Comment (Jens-H-Eriksen) [reply NuKeeper/Collaboration/CollaborationFactory.cs]: Yes... it was null, and is first set after this method is executed

File: Nukeeper.BitBucketLocal/BitBucketLocalPlatform.cs (hunk 2 above)
Comment (Jens-H-Eriksen) [reply Nukeeper.BitBucketLocal/BitBucketLocalPlatform.cs]: No... this removes any path in the URL

File: Nukeeper.BitBucketLocal/BitbucketLocalRestClient.cs (hunk 3 above)
Comment (Jens-H-Eriksen) [reply Nukeeper.BitBucketLocal/BitbucketLocalRestClient.cs]: I can do that

File: Nukeeper.BitBucketLocal/BitbucketLocalRestClient.cs (hunk 4 above)
Comment (Jens-H-Eriksen) [reply Nukeeper.BitBucketLocal/BitbucketLocalRestClient.cs]: All the default reviewers are added to the PR

File: Nukeeper.BitBucketLocal/Models/AddReviewer.cs (hunk 5 above)
Comment (Jens-H-Eriksen) [reply Nukeeper.BitBucketLocal/Models/AddReviewer.cs]: I agree

**Comment by shep1987** [comment]
LGTM

**Pull request merged**